### PR TITLE
chore: upgrade golangci-lint and set mem limits

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
Upgrade golangci-lint. This version is based on Go 1.19 which has [`GOMEMLIMIT`](https://weaviate.io/blog/2022/08/GOMEMLIMIT-a-Game-Changer-for-High-Memory-Applications.html). This can help us avoid CI OOM kills on `check` steps.
I set the number to `7GiB`, because it's run in Docker container with 8GB, so I want to give it a bit of space. Especially that this is not a hard limit (see the article).

The execution time on my machine is quite similar between `2GiB` and `7GiB`.

### Checklist prior to review

- [X] Link to docs PR or issue --
- [X] Link to UI issue or PR --
- [X] Is the [issue worked on linked][1]? --
- [X] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [X] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Unit Tests --
- [X] E2E Tests --
- [X] Manual Universal Tests --
- [X] Manual Kubernetes Tests --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by michaelbeaumont — COMMENTED** [review]
I think we can just run it once again, right? I.e. revert some of the changes here https://github.com/kumahq/kuma/pull/4907/files

**Comment by jakubdyszkiewicz** [comment]
Do you mean to run linters with one process?

**Comment by michaelbeaumont** [comment]
Yeah, I mean run `golangci-lint` once.

File: pkg/kds/mux/client.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func (c *client) startKDSMultiplex(ctx context.Context, log logr.Logger, conn *g
 		return
 	}
 	bufferSize := len(registry.Global().ObjectTypes())
+	// nolint:contextcheck
```
Comment (michaelbeaumont) [inline pkg/kds/mux/client.go:131]: Yeah, I think this comes because we break the "don't put context in a struct" rule in a lot of places

**Review by michaelbeaumont — APPROVED** [review]
:pray:

**Review by lukidzi — APPROVED** [review]

**Comment by jakubdyszkiewicz** [comment]
So setting `GOMEMLIMIT` still gets us OOM kills.

I changed the docker executor from large to xlarge. The cost raises from 20 tokens/min to 40, but the execution time is 30% faster so we pay just a little bit more (insignificant change when you compare this to other jobs), but we get:
* stability of not getting killed (xlarge has 16GB and lint is never a problem on such machines, we run 16GB setup in dev-linux).
* lower latency of CI pipeline, because E2E jobs depend on check.

**Pull request merged**